src/app/api/cronjob/route.ts

```diff
@@ -0,0 +1,177 @@
+  semesters,
+  weekdayEnum,
+} from "@/lib/db/schema";
+import { getWeekday } from "@/lib/utils";
+import { insertGameSessionSchema } from "@/lib/validators";
+import { Weekday } from "@/types/types";
+
+export async function POST(req: NextRequest) {
+  try {
+    // get current day, correct to NZ timezone and reset to 12am
+    let now: Date = new Date();
+    now = new Date(now.getTime() - now.getTimezoneOffset() * 60000);
+    now = new Date(now.getTime() - (now.getTime() % (3600 * 24 * 1000)));
+
+    // Testing date
+    //now = new Date(2024, 1, 21);
+    //now = new Date(2024, 1, 22);
+    //now = new Date(2024, 2, 28);
+    //now = new Date(2024, 2, 29);
+    //now = new Date(2024, 3, 7);
+    //now = new Date(2024, 3, 8);
+    //now = new Date(2024,5,24);
+    //now = new Date(2024,5,25);
+    //now = new Date(now.getTime() - now.getTimezoneOffset() * 60000);
+    //console.log(now);
+
+    const currentDateUTC = new Date(
+      now.getTime() + now.getTimezoneOffset() * 60000,
+    );
+
+    // Find current semester
+    const activeSemester = await db.query.semesters.findFirst({
+      where: and(
+        lte(
+          semesters.startDate,
+          format(addDays(currentDateUTC, 7), "yyyy-MM-dd"),
+        ),
+        gte(semesters.endDate, format(currentDateUTC, "yyyy-MM-dd")),
+      ),
+    });
+
+    if (!activeSemester) {
+      return NextResponse.json("No active semester found.", { status: 400 });
+    }
+
+    //const semesterStartDate = new Date(activeSemester.startDate);
+    const semesterEndDate = new Date(activeSemester.endDate);
+    const semesterBreakStart = new Date(activeSemester.breakStart);
+    const semesterBreakEnd = new Date(activeSemester.breakEnd);
+    const bookingOpenDay = activeSemester.bookingOpenDay;
+    const diff = findDiff(now, bookingOpenDay);
+    const startDate = addDays(currentDateUTC, diff);
+
+    // Check if the next batch of sessions is within the semester break or after the end of semester
+    if (startDate > semesterEndDate) {
+      return NextResponse.json(
+        "No sessions were generated as they are all after the semester end date.",
+        {
+          status: 400,
+        },
+      );
+    }
+
+    if (
+      addDays(startDate, 6) <= semesterBreakEnd &&
+      startDate >= semesterBreakStart
+    ) {
+      return NextResponse.json(
+        "No sessions were generated as they are all within the semester break.",
+        {
+          status: 400,
+        },
+      );
+    }
+
+    await db.transaction(async (tx) => {
+      // For each day in the session week, create a game session
+      for (let i: number = 0; i < 7; i++) {
+        const day = addDays(startDate, i);
+
+        if (day > semesterEndDate) {
+          break;
+        }
+
+        if (day >= semesterBreakStart && day <= semesterBreakEnd) {
+          continue;
+        }
+
+        //check for gameSessionScheduleExceptions and continue if deleted, and if edited then insert that instead and continue
```

Comment:
you can implement the gameSessionException now.

If an exception exists:
- if isDeleted is true, we can skip this.
- if isDeleted is false, we use the details on that to create the game session instead of the gameSessionSchedule